pulp_rpm/app/handler.py

```diff
@@ -0,0 +1,55 @@
+from tempfile import TemporaryDirectory
+
+from aiohttp.web_response import Response
+from django.conf import settings
+
+from pulpcore.app.settings import CONTENT_PATH_PREFIX
+from pulpcore.content import Handler
+from pulpcore.content.handler import PathNotResolved
+from pulp_rpm.app.models import RpmDistribution, RpmRepository
+from pulp_rpm.app.settings import RPM_CONTENT_PATH_PREFIX
+
+
+class RPMContentHandler(Handler):
+    distribution_model = RpmDistribution
+
+    def _match_permit_distribution(self, request, path):
+        distribution = self._match_distribution(path)
+        self._permit(request, distribution)
+        return distribution
+
+    async def _match_and_stream(self, path, request):
+        if path.endswith('/config.repo'):
+            distribution = self._match_permit_distribution(request, path)
+            val = f"""[{distribution.name}]
+enabled=1
+baseurl={settings.CONTENT_ORIGIN}{CONTENT_PATH_PREFIX}{distribution.base_path}
+gpgcheck=0
+"""
+            repository_pk = distribution.publication.repository.pk
+            repository = RpmRepository.objects.get(pk=repository_pk)
+            signing_service = repository.metadata_signing_service
+            if signing_service is None:
+                val += 'gpgrepocheck=0'
+            else:
+                val += f"""gpgrepocheck=1
+gpgkey={settings.CONTENT_ORIGIN}{RPM_CONTENT_PATH_PREFIX}/{distribution.base_path}/public.key
+"""
+            return Response(body=val)
+
+        if path.endswith('/public.key'):
```

Comment:
The public.key should already be present in a Publication so there doesn't need to be any handling of it in this code.

Reply:
By the looks of models, this data is not stored in the Publication.